Controller/Checkout/SecondChance.php

```diff
@@ -142,6 +142,11 @@ public function execute()
                     $customer       = $this->customerFactory->create()->load($customerId);
                     $sessionManager = $this->sessionFactory->create();
                     $sessionManager->setCustomerAsLoggedIn($customer);
+                } elseif ($customerEmail = $order->getCustomerEmail()) {
```

Comment:
I see there is a collection that is being iterated ... would like to understand what is the context - how it was meant to work, do you have multiple tokens/quotes, which one is restored? I can see that first customerId is checked and if that is not present, it tries the guest checkout flow... if you have multiple items in the collection - what should happen?

Reply:
Second chance collection iterated because looking for token in db. Second chance not save user information, i`ts load information from order, so in case BP-938 user was created after guest order with same email, and in this case order can`t load quote which use guest user and registered user